### DISCUSSION TITLE
How to resolve links in markdown

### DISCUSSION THREAD

**Discussion by PuneetPunamiya** [remarkjs/remark#558] [Q&A] [answered]
We have a UI which renders a readme from a repo, the repo has multiple directories and the readme points the directories using relative path so, is there anyway where we could get let the markdown link navigate to the proper page if the link has relative path

For e.g. 

We get the readme as ```[samples](../0.1/samples/secret.yaml)```

But it should navigate to https://github.com/tektoncd/catalog/aws-cli/0.1/samples/secret.yaml

**Comment by wooorm** [comment, marked as answer]
remark gives you a syntax tree. You can transform that syntax tree with plugins: https://unifiedjs.com/learn/guide/create-a-plugin/.

Here is an example of a plugin where I rewrite URLs from markdown in readmes on GitHub, so that it can be displayed on unifiedjs.com: https://github.com/unifiedjs/unifiedjs.github.io/blob/src/generate/plugin/rehype-resolve-urls.js.

**Reply by PuneetPunamiya** [reply]
@wooorm  Thanks a lot for the reply, I'll give it a try on the example you have referenced
